cmd/openshift-install/gather.go

```diff
@@ -204,6 +205,37 @@ func unSupportedPlatformGather(directory string) error {
 	return logGatherBootstrap(gatherBootstrapOpts.bootstrap, 22, gatherBootstrapOpts.masters, directory)
 }
 
+type operatorConditionType int
+
+const (
+	conditionUnknown     operatorConditionType = -1
+	conditionAvailable   operatorConditionType = 0
+	conditionDegraded    operatorConditionType = 1
+	conditionProgressing operatorConditionType = 2
+	conditionUpgradeable operatorConditionType = 3
+)
+
+type OperatorCondition struct {
+	priority         operatorConditionType
+	operatorName     string
+	conditionType    configv1.ClusterStatusConditionType
+	conditionStatus  configv1.ConditionStatus
+	conditionReason  string
+	conditionMessage string
+}
+
+type OperatorConditions []OperatorCondition
+
+func (o OperatorConditions) Len() int                 { return len(o) }
+func (o OperatorConditions) Less(i, j int) bool       { return o[i].priority > o[j].priority }
+func (o OperatorConditions) Swap(i, j int)            { o[i], o[j] = o[j], o[i] }
+func (o *OperatorConditions) Push(pushed interface{}) { *o = append(*o, pushed.(OperatorCondition)) }
+func (o *OperatorConditions) Pop() (popped interface{}) {
+	popped = (*o)[len(*o)-1]
+	*o = (*o)[:len(*o)-1]
+	return
+}
+
```

Comment:
hmm, personally i think heap is a little too heavy for what we need.

also since this gather is becoming too nuanced, can we move it to some `pkg/<>` style package?